algorithms/HPdclassifier/R/hpdsampling.R

```diff
@@ -0,0 +1,115 @@
+#                                                                                   
+# This program is free software; you can redistribute it and/or                     
+# modify it under the terms of the GNU General Public License                       
+# as published by the Free Software Foundation; either version 2                    
+# of the License, or (at your option) any later version.                            
+#                                                                                   
+# This program is distributed in the hope that it will be useful,                   
+# but WITHOUT ANY WARRANTY; without even the implied warranty of                    
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                      
+# GNU General Public License for more details.                                      
+#                                                                                   
+# You should have received a copy of the GNU General Public License                 
+# along with this program; if not, write to the Free Software                       
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.    
+####################################################################################
+
+####################################################################################
+#  File hpdsampling.R
+#  This code is a distributed version of sample availabel in R gbm package.
+
+####################################################################################
+
+
+
+####################################################################################
+#                  Distributed sampling
+####################################################################################
+hpdsampling <- function(dfX,daY, Ns, npartition)
+{
+  #dfX: predictor variables in dframe/darray
+  #daY: vector of outcomes in darray (dvector/dfactor is not supported yet in2015-7-01
+  #npartition: now the sampled data have the same npartition as input. In the future, the input and output may have different partitions
+  #Ns : size of sub-chunks. #Ns <- 1*ceiling((nTrain/npartition)/npartition) # size of sampled sub-chunk
+  #function output: dfRX, daRY                 
+
+  ####################################################################################
+  # check function arguments
+   if(missing(dfX))
+	stop("'dfX' is a required argument")
+
+   if(missing(daY))
+	stop("'Y_train' is a required argument")
+
+   if(missing(Ns))
+	stop("'Ns' is a required argument")
+
+   if(missing(npartition))
+	stop("'npartition' is a required argument")
+
+   if(!is.dframe(dfX) && !is.darray(dfX))
+       stop("'dfX' must be a dframe or darray")
+
+   # for binary classification/regression, daY is a darray. For multinomial distribution (multi-classification), daY is a dframe (factor vecotr originally)
+   if(!is.dframe(daY) && !is.darray(daY))
+       stop("'daY' must be a dframe or darray")
+
```

Comment:
What happens when dfX and dfY have different number of rows? What when they are dobjects with the same number of rows but different number of partitions? Or even the same number of partitions, but inconsistent partitions in terms of row number?